framework/include/materials/MaterialData.h

```diff
@@ -275,20 +278,28 @@ MaterialData::getPropertyHelper(const std::string & prop_name,
   // Register/get the ID of the property
   const auto prop_id = addPropertyHelper(
       prop_name, typeid(T), state, declare ? &castRequestorToDeclarer(requestor) : nullptr);
+  const auto size = prop_id + 1;
+
+  // get the maximum state id requested for this property so far
+  if (_max_state_requested.size() < size)
+    _max_state_requested.resize(size, 0);
```

Comment:
This data should already exist in `MaterialPropertyStorage`:

```
_storage.getPropRecord(prop_id).state
```

Reply:
Unfortunately you'll need to add it to the source file due to includes...

Reply:
Circular dependency. Refactoring that will be a pain.

Reply:
No, easy. I put a function to get that value in MaterialData.C. That might be what you meant in the first place.

Reply:
Yep, that's what I meant